streams/src/main/java/org/apache/kafka/streams/state/internals/StoreQueryUtils.java

```diff
@@ -346,6 +346,23 @@ public static <V> Function<byte[], V> getDeserializeValue(final StateSerdes<?, V
         return byteArray -> deserializer.deserialize(serdes.topic(), byteArray);
     }
 
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    public static <V> Function<byte[], V> getDeserializeValue2(final StateSerdes<?, V> serdes,
+                                                               final StateStore wrapped,
+                                                               final boolean isDSLStore ) {
+        final Serde<V> valueSerde = serdes.valueSerde();
+        final boolean timestamped = WrappedStateStore.isTimestamped(wrapped) || isDSLStore;
```

Comment:
The issue is actually inside `isTimestamped` -- for the `KeyValueToTimestampedKeyValueByteStoreAdapter` it incorrectly returns `false` while it should return `true`. The root cause is a bug in `KeyValueToTimestampedKeyValueByteStoreAdapter` itself: it does not implement the marker interface `TimestampedBytesStore` but it should.

Reply:
ok